test/Microsoft.AspNetCore.Razor.Language.Test/CodeGeneration/DesignTimeDirectiveTargetExtensionTest.cs

```diff
@@ -0,0 +1,212 @@
+        }
+
+        [Fact]
+        public void WriteDesignTimeDirective_WithNamespaceToken_WritesLambda()
+        {
+            // Arrange
+            var extension = new DesignTimeDirectiveTargetExtension();
+            var context = new CSharpRenderingContext()
+            {
+                Writer = new CSharpCodeWriter()
+            };
+
+            var node = new DesignTimeDirectiveIRNode();
+            var token = new DirectiveTokenIRNode()
+            {
+                Content = "System.Collections.Generic",
+                Descriptor = new DirectiveTokenDescriptor()
+                {
+                    Kind = DirectiveTokenKind.Namespace
+                }
+            };
+            node.Children.Add(token);
+
+            // Act
+            extension.WriteDesignTimeDirective(context, node);
+
+            // Assert
+            var csharp = context.Writer.Builder.ToString();
+            Assert.Equal(
+@"#pragma warning disable 219
+private void __RazorDirectiveTokenHelpers__() {
+((System.Action)(() => {
+global::System.Object __typeHelper = nameof(System.Collections.Generic);
+}
+))();
+}
+#pragma warning restore 219
+",
+                    csharp,
+                    ignoreLineEndingDifferences: true);
+        }
+
+        [Fact]
+        public void WriteDesignTimeDirective_WithMemberToken_WritesLambda()
+        {
+            // Arrange
+            var extension = new DesignTimeDirectiveTargetExtension();
+            var context = new CSharpRenderingContext()
+            {
+                Writer = new CSharpCodeWriter()
+            };
+
+            var node = new DesignTimeDirectiveIRNode();
+            var token = new DirectiveTokenIRNode()
+            {
+                Content = "Foo",
+                Descriptor = new DirectiveTokenDescriptor()
+                {
+                    Kind = DirectiveTokenKind.Member
+                }
+            };
+            node.Children.Add(token);
+
+            // Act
+            extension.WriteDesignTimeDirective(context, node);
+
+            // Assert
+            var csharp = context.Writer.Builder.ToString();
+            Assert.Equal(
+@"#pragma warning disable 219
+private void __RazorDirectiveTokenHelpers__() {
+((System.Action)(() => {
+global::System.Object Foo = null;
+}
+))();
+}
+#pragma warning restore 219
+",
+                    csharp,
+                    ignoreLineEndingDifferences: true);
+        }
+
+        [Fact]
+        public void WriteDesignTimeDirective_WithStringToken_WritesLambda()
+        {
+            // Arrange
+            var extension = new DesignTimeDirectiveTargetExtension();
+            var context = new CSharpRenderingContext()
+            {
+                Writer = new CSharpCodeWriter()
+            };
+
+            var node = new DesignTimeDirectiveIRNode();
+            var token = new DirectiveTokenIRNode()
+            {
+                Content = "Value",
```

Comment:
I intentionally did this in a single test to verify code gen with more than one child.